### PR TITLE
[ADD] account_banking_reconciliation

### PR DESCRIPTION
**Pull request by max3903**
# Todo

- [x] Migrate code to version 10
- [x] Convert report to QWeb
- [x] Update the documentation

#173

### CODE REVIEW

File: account_banking_reconciliation/views/report_bank_statement_summary.xml [hunk 1]
```diff
@@ -0,0 +1,190 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+<data>
```
Comment (max3903) [inline account_banking_reconciliation/views/report_bank_statement_summary.xml]: @smangukiya You can remove the `<data>` tag.

**Comment by max3903** [comment]
@smangukiya You can find the problems causing the checks to fail here:
https://travis-ci.org/OCA/bank-statement-reconcile/jobs/237818444#L525

**Comment by smangukiya** [comment]
@max3903 : Please review it. Thanks!

File: account_banking_reconciliation/models/account_banking_reconciliation.py [hunk 2]
```diff
@@ -0,0 +1,661 @@
...
+                    if dr_item.move_line_id and dr_item.cleared_bank_account:
+                        refdict[dr_item.move_line_id.id] = \
+                            dr_item.cleared_bank_account
+
+        # for the statement
+        for statement in self:
+
+            # process only if the statement is in draft state
+            if statement.state == 'draft':
+                vals = self.onchange_account_id(
```
Comment (AaronHForgeFlow) [inline account_banking_reconciliation/models/account_banking_reconciliation.py]: onchange_account_id() takes exactly 1 argument (4 given)

Comment (AaronHForgeFlow) [reply account_banking_reconciliation/models/account_banking_reconciliation.py]: Just `vals = statement.onchange_account_id()`

File: account_banking_reconciliation/models/account_banking_reconciliation.py [hunk 3]
```diff
@@ -0,0 +1,661 @@
...
+                                      "reconciled.")
+    ending_date = fields.Date('Ending Date', required=True,
+                              states={'done': [('readonly', True)]},
+                              default=time.strftime('%Y-%m-%d'),
+                              help="The ending date of your bank statement.")
+    last_ending_date = fields.Date('Last Stmt Date',
+                                   help="The previous statement date "
+                                        "of your bank statement.")
+    starting_balance = fields.Float('Starting Balance', required=True,
+                                    digits_compute=dp.get_precision('Account'),
```
Comment (AaronHForgeFlow) [inline account_banking_reconciliation/models/account_banking_reconciliation.py]: `digits_compute` is deprecated use `digits` instead

File: account_banking_reconciliation/models/account_banking_reconciliation.py [hunk 4]
```diff
@@ -0,0 +1,661 @@
...
+                                    help="The Starting Balance on your "
+                                         "bank statement.",
+                                    states={'done': [('readonly', True)]})
+    ending_balance = fields.Float('Ending Balance', required=True,
+                                  digits_compute=dp.get_precision('Account'),
```
Comment (AaronHForgeFlow) [inline account_banking_reconciliation/models/account_banking_reconciliation.py]: same here

File: account_banking_reconciliation/views/account_banking_reconciliation.xml [hunk 5]
```diff
@@ -0,0 +1,281 @@
...
+              id="bank_acc_rec_statement_menu"
+              parent="account.menu_finance_entries"
+              sequence="5"/>
+
+    <act_window id="act_account_move_line_to_add" name="Journal Items"
+                res_model="account.move.line"
+                src_model="bank.acc.rec.statement"/>
+
+    <act_window id="act_supplier_payment_to_add" name="Supplier Payment"
+                domain="[('journal_id.type', 'in', ['bank', 'cash']), ('type','=','receipt')]"
```
Comment (AaronHForgeFlow) [inline account_banking_reconciliation/views/account_banking_reconciliation.xml]: invalid leaf type, please use `voucher_type` instead

File: account_banking_reconciliation/views/account_banking_reconciliation.xml [hunk 6]
```diff
@@ -0,0 +1,281 @@
...
+                context="{'type':'payment'}" res_model="account.voucher"
+                src_model="bank.acc.rec.statement"/>
+
+    <act_window id="act_customer_payment_to_add" name="Customer Payment"
+                domain="[('journal_id.type', 'in', ['bank', 'cash']), ('type','=','receipt')]"
```
Comment (AaronHForgeFlow) [inline account_banking_reconciliation/views/account_banking_reconciliation.xml]: same here

File: account_banking_reconciliation/models/account_banking_reconciliation.py [hunk 7]
```diff
@@ -0,0 +1,661 @@
...
+            for statement_line in statement_lines:
+                statement_line_ids.append(statement_line)
+
+                if statement_line.move_line_id:
+                    # Find move lines related to statement lines
+                    line_ids.append(
+                        statement_line.move_line_id)
+
+            # Reset 'Cleared' and 'Bank Acc Rec Statement ID' to False
+            line_ids.write({'cleared_bank_account': False,
```
Comment (AaronHForgeFlow) [inline account_banking_reconciliation/models/account_banking_reconciliation.py]: This is wrong, line_ids is a list, so is statement_line_ids

File: account_banking_reconciliation/views/account_banking_reconciliation.xml [hunk 8]
```diff
@@ -0,0 +1,281 @@
...
+                            <label for='account_label'
+                                   string='Deposits, Credits, and Interest'/>
+                            <field colspan="4" mode="tree"
+                                   name="debit_move_line_ids"
+                                   nolabel="1" widget="one2many_list"
+                                   height="300">
+                                <form string="Deposits, Credits, and Interest"
+                                      version="7.0">
+                                    <field name="cleared_bank_account"/>
+                                    <field name="date" readonly="1"/>
```
Comment (aheficent) [inline account_banking_reconciliation/views/account_banking_reconciliation.xml]: maybe it'd be nice to allow to create manual adjusting journal entries here to advisers, not sure thought

File: account_banking_reconciliation/models/account_banking_reconciliation.py [hunk 9]
```diff
@@ -0,0 +1,661 @@
...
+        return True
+
+    @api.multi
+    def action_select_all(self):
+        """Mark all the statement lines as 'Cleared'."""
+        for statement in self:
+            statement_lines = \
+                statement.credit_move_line_ids + statement.debit_move_line_ids
+            statement_line_ids = map(lambda x: x.id, statement_lines)
+            statement_line_ids.write({'cleared_bank_account': True})
```
Comment (AaronHForgeFlow) [inline account_banking_reconciliation/models/account_banking_reconciliation.py]: `statement_line_ids` is a list not object list

**Comment by bodedra** [comment]
@aheficent Thanks for your notice issues. Your valuable comments are attended! Would you please review it again ?

**Comment by bodedra** [comment]
@aheficent friendly reminder. :)

**Review by bodedra — APPROVED** [review]
LGTM

**Pull request merged**